python/tvm/auto_scheduler/search_task.py

```diff
@@ -63,6 +66,11 @@ def __getstate__(self):
     def __setstate__(self, state):
         self.dag = state["dag"]
         self.workload_key = state["workload_key"]
+        try:
+            func_name = json.loads(self.workload_key)[0]
+        except Exception:  # pylint: disable=broad-except
+            raise RuntimeError("Invalid workload key %s" % self.workload_key)
+        register_workload_tensors(func_name, self.dag.tensors)
```

Comment:
We should do more checks here.
There are two kinds of workloads.
You can find the note here 
https://github.com/apache/incubator-tvm/blob/51d81fb71781eb606f4b563ec28290dbd8d04faf/python/tvm/auto_scheduler/workload_registry.py#L42-L50

We have to handle two cases correctly without incorrect overwriting

Reply:
Changed the logic accordingly. Now we only register workloads extracted from Relay programs.